src/utils/date.ts

```diff
@@ -10,3 +12,63 @@ export const formatDate = (date: string | Date, formatStr = DEFAULT_DATE_FORMAT)
     throw new Error(error.message);
   }
 };
+
+type Asset = {
+  datetime?: string;
+  created?: string;
+  updated?: string;
+  start_datetime?: string;
+  end_datetime?: string;
+};
+
+type StacData = Collection | StacItem;
+
+export type ExtractedDates = {
+  start: string | null;
+  end: string | null;
+  datetime: string | null;
+  created: string | null;
+  updated: string | null;
+  start_datetime: string | null;
+  end_datetime: string | null;
+};
+
+export const extractDates = (stacData: StacData): ExtractedDates => {
+  const dates: ExtractedDates = {
+    start: null,
+    end: null,
+    datetime: null,
+    created: null,
+    updated: null,
+    start_datetime: null,
+    end_datetime: null,
+  };
+
+  // Handle Collection temporal extent
+  if (stacData.type === 'Collection') {
+    const temporalExtent = stacData?.extent?.temporal?.interval?.[0]; // First interval
+    if (temporalExtent) {
+      dates.start = temporalExtent[0] || null;
+      dates.end = temporalExtent[1] || null;
+    }
+  }
+
+  // Handle Item datetime in properties
+  if (stacData.type === 'Feature') {
+    dates.start = stacData.properties.datetime || stacData.properties.start_datetime || null;
+    dates.end = stacData.properties.end_datetime || null;
+  }
+
+  // Handle assets (in both Collection and Item)
+  if (stacData.assets) {
+    Object.values(stacData.assets).forEach((asset: Asset) => {
+      dates.datetime = asset.datetime || dates.datetime;
```

Comment:
Is this overwriting timestamps for the record as a whole with ones for particular assets?

If we had, say, a Sentinel 2 collection with a date range of 2015-2024, plus a logo/thumbnail asset date 2023 then we should be careful not to mix these.

For some items it might in theory be possible to work out a missing date range or timestamp from the assets but I think this would be rather error-prone as we'd have to identify which asset is the right one.

Reply:
I have removed this part of the method. When searching for dates on a STAC object, it will not check for the dates on any of its assets